### PR TITLE
To fix flaky downward api volume e2e

### LINKED ISSUE #17874: e2e flake: Downward API volume should update labels on modification [Conformance]
In parallel suite:

```/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/downwardapi_volume.go:108 Timed out after 15.000s. Expected : content of file "/etc/labels": key1="value1" key2="value2" content of file "/etc/labels": key1="value1" key2="value2" content of file "/etc/labels": key1="value1" key2="value2" content of file "/etc/labels": key1="value1" key2="value2" content of file "/etc/labels": key1="value1" key2="value2" to contain substring : key3="value3" ```

@thockin @ixdy @wojtek-t @mikedanese @erictune

### PR DESCRIPTION
**Pull request by sdminonne**
To fix https://github.com/kubernetes/kubernetes/issues/17874

### CODE REVIEW

**Comment by sdminonne** [comment]
@pmorie as discussed via IRC

**Comment by ihmccreery** [comment]
Reassigning to @pmorie, since it sounds like you have context here.

File: test/e2e/downwardapi_volume.go [hunk 1]
```diff
@@ -81,11 +83,30 @@ var _ = Describe("Downward API volume", func() {
 		},
 			podLogTimeout, poll).Should(ContainSubstring("key1=\"value1\"\n"))
 
-		//modify labels
-		pod.Labels["key3"] = "value3"
-		_, err = f.Client.Pods(f.Namespace.Name).Update(pod)
-		Expect(err).NotTo(HaveOccurred())
-
+		//modify labels	in a update/get loop
```
Comment (caesarxuchao) [inline test/e2e/downwardapi_volume.go]: If we know the conflicts will be innocuous, we can set pod.ResourceVersion="" before passing it to Update(), that will force an update.

File: test/e2e/downwardapi_volume.go [hunk 2]
```diff
@@ -81,11 +83,30 @@ var _ = Describe("Downward API volume", func() {
...
+					return false, fmt.Errorf("failed to get pod: %v", err)
+				}
+			}
+			pod.Labels["key3"] = "value3"
+			pod, err = f.Client.Pods(f.Namespace.Name).Update(pod)
+			if err == nil {
+				Logf("Successfully updated pod")
+				return true, nil
+			}
+			if errors.IsConflict(err) {
```
Comment (pmorie) [inline test/e2e/downwardapi_volume.go]: i think if we don't do it now we will probably eventually add a loop for retries -- want to make it a loop w/ 2 retries for now?

Comment (pmorie) [inline test/e2e/downwardapi_volume.go]: Disregard above comment, more closely read @caesarxuchao's comment :)

File: test/e2e/downwardapi_volume.go (hunk 1 above)
Comment (pmorie) [inline test/e2e/downwardapi_volume.go]: @sdminonne Do you know what the source of the conflict was at all?

Comment (pmorie) [inline test/e2e/downwardapi_volume.go]: I think in general it should be okay to force the update for this test -- there should be no other actor material to the test updating the pod.

**Comment by sdminonne** [comment]
@pmorie @caesarxuchao. Going to modify to force the update with `pod.ResourceVersion=""`

**Comment by sdminonne** [comment]
@pmorie @caesarxuchao PTAL

**Pull request merged**